src/ophyd_async/epics/areadetector/single_trigger_det.py

```diff
@@ -20,12 +26,13 @@ def __init__(
     ) -> None:
         self.drv = drv
         self.__dict__.update(plugins)
-        self.set_readable_signals(
-            # Can't subscribe to read signals as race between monitor coming back and
-            # caput callback on acquire
-            read_uncached=[self.drv.array_counter] + list(read_uncached),
-            config=[self.drv.acquire_time],
+
+        self.add_readables(
+            HintedSignal.uncached, self.drv.array_counter, *read_uncached
         )
+
+        self.add_readables(ConfigSignal, self.drv.acquire_time)
```

Comment:
This would have to be keyword
```suggestion
        self.add_readables(self.drv.acquire_time, wrapper=ConfigSignal)
```